### DISCUSSION TITLE
FleetDM - Server IP address changed but Fleet is not recognizing it

### DISCUSSION THREAD

**Discussion by Nolan-Mcfly** [Security-Onion-Solutions/securityonion#4565] [General]
The issue I am having with Fleet is that I originally installed Security Onion with the IP address of `192.168.0.167`, but I changed my IP to `192.168.0.19` as a static address. .167 was not a static IP for the server at the beginning of the installation. Everything is working, as I ran `so-ip-update`, but Fleet is not showing any host names online besides the server. I have checked a few different config files, but I cannot find anything that would point to Fleet not working with `192.168.0.19`, in fact the web page for Fleet is loading at the .19 address. I have rebooted, restarted Fleet with `so-fleet-restart --force` and when I run that command, I get this among the output:

----------
          ID: fleetdb
    Function: mysql_database.present
        Name: fleet
      Result: False
     Comment: MySQL Error 1130: Host '192.168.0.19' is not allowed to connect to this MySQL server
     Started: 10:34:47.740950
    Duration: 202.909 ms
     Changes:   
----------
          ID: fleetdbuser
    Function: mysql_user.present
      Result: False
     Comment: One or more requisite failed: fleet.fleetdb
     Started: 10:34:47.946236
    Duration: 0.009 ms
     Changes:   
----------
          ID: fleetdbpriv
    Function: mysql_grants.present
      Result: False
     Comment: One or more requisite failed: fleet.fleetdb
     Started: 10:34:47.947889
    Duration: 0.009 ms
     Changes:   
----------

I am not sure how to go about fixing this, aside from just using `192.168.0.167` as my static IP. Any and all thoughts or suggestions are welcome, thank you.

**Comment by defensivedepth** [comment]
@Nolan-Mcfly Can you please post the output from the following (sanitizing as needed)? 

`sudo salt-call pillar.get global`

**Reply by Nolan-Mcfly** [reply]
@defensivedepth here is the output from the command:

```local:
    ----------
    airgap:
        False
    cortexkey:
        ##################
    cortexorgname:
        SecurityOnion
    cortexorguserkey:
        ##################
    cortexplaysecret:
        ##################
    dockernet:
        172.17.0.0
    fleet_custom_hostname:
        None
    fleet_hostname:
        ubuntu-desktop
    fleet_ip:
        192.168.0.167
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2021-06-14-20:16
    fleet_packages-version:
        2
    hivekey:
        ##################
    hiveplaysecret:
        ##################
    hnmanager:
        192.168.0.0/24
    ids:
        Suricata
    imagerepo:
        security-onion-solutions
    managerip:
        192.168.0.167
    mdengine:
        SURICATA
    pipeline:
        redis
    sensoronikey:
        ##################
    soversion:
        2.3.52
    url_base:
        192.168.0.167
    wazuh:
        1

**Reply by defensivedepth** [reply]
@Nolan-Mcfly Can you post the output of the following? `sudo salt-call grains.item ip_interfaces`

Also do a  `so-fleet-restart --force` one more time and confirm that you see the same error.

**Reply by Nolan-Mcfly** [reply]
@defensivedepth Before I do this, I assume I should change my IP to the desired 192.168.0.19, run `so-ip-update` and then run these commands, correct?

**Reply by defensivedepth** [reply]
@Nolan-Mcfly Not sure I understand. In the original post, you said `so-ip-update` had already been run, correct? 

192.168.0.167 is the old IP, correct? So it looks like the `so-ip-update` didn't work?

**Reply by Nolan-Mcfly** [reply]
@defensivedepth I apologize for the confusion, after the initial post I took it back to `192.168.0.167` to be able to continue chugging along at configuring fleet and the other tools to keep everything working until I could try to take it back to `192.168.0.19`. Since that reply I made to you yesterday, I have taken my IP back to `192.168.0.19` through `so-ip-update` and I will post the output of the commands here:

`sudo salt-call pillar.get global`

```local:
    ----------
    airgap:
        False
    cortexkey:
        ######################
    cortexorgname:
        SecurityOnion
    cortexorguserkey:
        ######################
    cortexplaysecret:
        ######################
    dockernet:
        172.17.0.0
    fleet_custom_hostname:
        None
    fleet_hostname:
        ubuntu-desktop
    fleet_ip:
        192.168.0.19
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2021-06-14-20:16
    fleet_packages-version:
        2
    hivekey:
        ######################
    hiveplaysecret:
        ######################
    hnmanager:
        192.168.0.0/24
    ids:
        Suricata
    imagerepo:
        security-onion-solutions
    managerip:
        192.168.0.19
    mdengine:
        SURICATA
    pipeline:
        redis
    sensoronikey:
        ######################
    soversion:
        2.3.52
    url_base:
        192.168.0.19
    wazuh:
        1
```

`sudo salt-call grains.item ip_interfaces`

```
local:
    ----------
    ip_interfaces:
        ----------
        bond0:
        docker0:
            - 172.17.0.1
        eno1:
            - 192.168.0.19
            - fe80::10e0:2f54:9cc3:d0dc
        eno2:
        lo:
            - 127.0.0.1
        veth00d4bd3:
        veth06aca0a:
        veth2584c39:
        veth2dacf21:
        veth2f184cd:
        veth32bd343:
        veth3e41b44:
        veth40710b2:
        veth57c20aa:
        veth5d9f763:
        veth605e9a1:
        veth625d719:
        veth6d8ac86:
        veth72b1af7:
        veth73dc209:
        veth74562d0:
        veth8afeec3:
        veth8ecf79e:
        veth9599804:
        veth973862f:
        vetha60ccec:
        vethbccd9c1:
        vethc8104fd:
        vethcf0417f:
        vethd4cda8c:
        vethdc813ca:
        vethdddd2f4:
        vethdf49609:
        vethebdc6ae:
```

`sudo so-fleet-restart --force`

```
local:
    - Signal 9 sent to job 20210708134613135288 at pid 45695
so-fleet
so-fleet
[ERROR   ] MySQL Error 1130: Host '192.168.0.19' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '192.168.0.19' is not allowed to connect to this MySQL server
local:
----------
          ID: mysqlpkgs
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 09:57:47.468736
    Duration: 304.674 ms
     Changes:   
----------
          ID: mysqletcdir
    Function: file.directory
        Name: /opt/so/conf/mysql/etc
      Result: True
     Comment: The directory /opt/so/conf/mysql/etc is in the correct state
     Started: 09:57:47.793192
    Duration: 20.957 ms
     Changes:   
----------
          ID: mysqlpiddir
    Function: file.directory
        Name: /opt/so/conf/mysql/pid
      Result: True
     Comment: The directory /opt/so/conf/mysql/pid is in the correct state
     Started: 09:57:47.815923
    Duration: 6.07 ms
     Changes:   
----------
          ID: mysqletcsync
    Function: file.recurse
        Name: /opt/so/conf/mysql/etc
      Result: True
     Comment: The directory /opt/so/conf/mysql/etc is in the correct state
     Started: 09:57:47.822373
    Duration: 226.181 ms
     Changes:   
----------
          ID: mysqllogdir
    Function: file.directory
        Name: /opt/so/log/mysql
      Result: True
     Comment: The directory /opt/so/log/mysql is in the correct state
     Started: 09:57:48.049018
    Duration: 3.32 ms
     Changes:   
----------
          ID: mysqldatadir
    Function: file.directory
        Name: /nsm/mysql
      Result: True
     Comment: The directory /nsm/mysql is in the correct state
     Started: 09:57:48.052534
    Duration: 2.141 ms
     Changes:   
----------
          ID: so-mysql
    Function: docker_container.running
      Result: True
     Comment: Container 'so-mysql' is already configured as specified
     Started: 09:57:48.058227
    Duration: 3955.215 ms
     Changes:   
----------
          ID: so-mysql
    Function: cmd.run
        Name: until nc -z 192.168.0.19 3306; do sleep 1; done
      Result: True
     Comment: State was not run because none of the onchanges reqs changed
     Started: 09:57:52.028363
    Duration: 0.017 ms
     Changes:   
----------
          ID: so-mysql
    Function: module.run
      Result: True
     Comment: State was not run because none of the onchanges reqs changed
     Started: 09:57:52.030835
    Duration: 0.006 ms
     Changes:   
----------
          ID: append_so-mysql_so-status.conf
    Function: file.append
        Name: /opt/so/conf/so-status/so-status.conf
      Result: True
     Comment: File /opt/so/conf/so-status/so-status.conf is in correct state
     Started: 09:57:52.031001
    Duration: 20.786 ms
     Changes:   
----------
          ID: fleetcdir
    Function: file.directory
        Name: /opt/so/conf/fleet/etc
      Result: True
     Comment: The directory /opt/so/conf/fleet/etc is in the correct state
     Started: 09:57:52.052129
    Duration: 2.468 ms
     Changes:   
----------
          ID: fleetpackcdir
    Function: file.directory
        Name: /opt/so/conf/fleet/packs
      Result: True
     Comment: The directory /opt/so/conf/fleet/packs is in the correct state
     Started: 09:57:52.054790
    Duration: 2.246 ms
     Changes:   
----------
          ID: fleetnsmdir
    Function: file.directory
        Name: /nsm/osquery/fleet
      Result: True
     Comment: The directory /nsm/osquery/fleet is in the correct state
     Started: 09:57:52.057229
    Duration: 2.215 ms
     Changes:   
----------
          ID: fleetpacksync
    Function: file.recurse
        Name: /opt/so/conf/fleet/packs
      Result: True
     Comment: The directory /opt/so/conf/fleet/packs is in the correct state
     Started: 09:57:52.059650
    Duration: 522.003 ms
     Changes:   
----------
          ID: fleetpackagessync
    Function: file.recurse
        Name: /opt/so/conf/fleet/packages
      Result: True
     Comment: The directory /opt/so/conf/fleet/packages is in the correct state
     Started: 09:57:52.582271
    Duration: 2798.743 ms
     Changes:   
----------
          ID: fleetlogdir
    Function: file.directory
        Name: /opt/so/log/fleet
      Result: True
     Comment: The directory /opt/so/log/fleet is in the correct state
     Started: 09:57:55.381512
    Duration: 4.33 ms
     Changes:   
----------
          **ID: fleetdb
    Function: mysql_database.present
        Name: fleet
      Result: False
     Comment: MySQL Error 1130: Host '192.168.0.19' is not allowed to connect to this MySQL server
     Started: 09:57:55.388631
    Duration: 23.999 ms
     Changes:   
----------
          ID: fleetdbuser
    Function: mysql_user.present
      Result: False
     Comment: One or more requisite failed: fleet.fleetdb
     Started: 09:57:55.416386
    Duration: 0.007 ms
     Changes:   
----------
          ID: fleetdbpriv
    Function: mysql_grants.present
      Result: False
     Comment: One or more requisite failed: fleet.fleetdb
     Started: 09:57:55.418078
    Duration: 0.006 ms
     Changes:**   
----------
          ID: so-fleet
    Function: docker_container.running
      Result: True
     Comment: Created container 'so-fleet'
     Started: 09:57:55.419007
    Duration: 12433.748 ms
     Changes:   
              ----------
              container_id:
                  ----------
                  added:
                      ############################################################
              state:
                  ----------
                  new:
                      running
                  old:
                      None
----------
          ID: append_so-fleet_so-status.conf
    Function: file.append
        Name: /opt/so/conf/so-status/so-status.conf
      Result: True
     Comment: File /opt/so/conf/so-status/so-status.conf is in correct state
     Started: 09:58:07.857309
    Duration: 30.243 ms
     Changes:

Summary for local
-------------
Succeeded: 18 (changed=1)
Failed:     3
-------------
Total states run:     21
Total run time:   20.359 s
```

**Reply by defensivedepth** [reply]
With the new IP in place, can you run the following and let me know the results?

`sudo docker exec -it -u 0 so-mysql bash`
`echo $MYSQL_ROOT_HOST`

**Reply by Nolan-Mcfly** [reply]
@defensivedepth Those commands brought back `192.168.0.19`.

**Reply by defensivedepth** [reply]
ok, so the issue is that the mysql root user doesn't have permission to login from the new IP. I have created an issue to track this, so that we can get `so-ip-update` edited (https://github.com/Security-Onion-Solutions/securityonion/issues/4811)

In the meantime, you can change it manually:

-- Make note of the mysql root password
`sudo salt-call pillar.get secrets `

 -- Login to the MySQL container
`sudo docker exec -it -u 0 so-mysql bash`

-- Login to the mysql cli
`mysql -uroot -p`

-- Grant the root user the ability to login from the new IP
`GRANT ALL ON *.* to root@'192.168.0.19' IDENTIFIED BY '<Your MySQL Root Password>';`

-- Verify that there is a new Grant for the root user 
`select * from information_schema.user_privileges;`

**Reply by Nolan-Mcfly** [reply]
Ok, this did in fact fix the issue of so-fleet not loading because it cannot access MySQL. However, the issue of host machines in Fleet all showing as offline is persistent. Right now, the only host showing as online is my Ubuntu-Server, aka the host machine for the Fleet server. I did some investigating on one of the host machines, and found the old IP address `(192.168.0.167)` in the launcher FLAGS file in the path `C:\Program Files\Kolide\Launcher-so-launcher\conf.` I changed the IP address in that config file to 192.168.0.19 and then reran the launchers for Fleet and OSQuery, thinking it may resolve the issue, but it did not. Is there any need for changing the enroll secret? Should I re-setup the Fleet server/is there a way to change the flag file at the "add host" screen on Fleet?

**Reply by defensivedepth** [reply]
Didnt get a notification that you had replied!

Are you still having issues with it?

**Reply by Nolan-Mcfly** [reply]
@defensivedepth No problem, I should have followed up again but I'm sure you are very busy. Yes, it is still an issue. I tried a few other things like editing the conf file on a couple of the machines that are connected to Fleet because they still had the old Server IP and that didn't seem to fix the issue. Basically where I am at right now with this is that everything is working besides the hosts showing as online in Fleet, every host shows offline besides the Fleet server (host machine). I'm wondering if there's a way to get a new flag file and enroll secret if those two things are necessary to begin working again?

Thanks,

Nolan

**Reply by defensivedepth** [reply]
I think it might be a cert issue. 

Can you review the logs in the Application eventlog channel on one of the Windows systems, looking for `Launcher` logs? Best thing to do would be to restart the launcher service, which will force it to try and connect, and you should see connection error logs in the Application channel.

**Reply by mckinleyc** [reply]
I am currently having the same issue with all nodes showing offline. I cannot get any agent to connect and even the manager is showing offline. The cert is different between what is on the server and what is downloaded from the web interface. Could this be an issue?

**Reply by defensivedepth** [reply]
@mckinleyc Please open a new Discussion following these Guidelines - https://github.com/Security-Onion-Solutions/securityonion/discussions/1720

Thanks
